### DISCUSSION TITLE
Creating a objects within objects with a loop

### DISCUSSION THREAD

**Discussion by KojoBailey** [nlohmann/json#4291] [Q&A]
To put this simply, I have an `std::vector<std::string>` with values I don't know and a count of items I don't know.

For example, this vector's values could be `{ "one", "two", "three", "four" }`.

I want to have a loop that automatically creates a "chain" of objects-within-objects from this vector. Using the example vector from above, it should output the JSON:
```json
{
  "one": {
    "two": {
      "three": {
        "four"
      }
    }
  }
}
```

Afterwards, I should be able to merge this JSON with other JSON generated from other vectors.

For example, if I have another vector which is `{ "one", "two", "three", "bazinga" }`, they should merge together to form:
```json
{
  "one": {
    "two": {
      "three": {
        "four",
        "bazinga"
      }
    }
  }
}
```

At least, that's what I understand from JSON. Maybe what I've outlined so far is wrong, in which case please tell me.

So, what I'm stuck on is the actual looping part. In my head, this is what I'm expecting from the above JSON (conceptualised in C++):
```cpp
json_data["one"]["two"]["three"] = {"four", "bazinga"};
```
Although this creates an array (`[]` instead of `{}`) for `"four"` and `"bazinga"`, which I don't want... I think? To be honest I'm not entirely sure how to go about achieving what I want. All I have at the moment is a vision and no clue where to start lol

Any help? I can elaborate if need be

<details>
<summary>More specific contextual information...</summary>
I'm trying to create JSON from folder directories that lists all the files, folders, sub-folders, etc. in JSON format. Maybe there's a better solution to what I'm trying to achieve though than JSON? I just figured JSON would be the easiest to work with, but maybe I'm oblivious to something more vanilla and suited for C++ natively. I want to be able to access the data afterwords to construct custom MediaWiiki syntax stuff (not really relevant to my question though).
</details>

**Comment by gregmarr** [comment]
```
{
  "one": {
    "two": {
      "three": {
        "four"
      }
    }
  }
}
```
This is not valid JSON.  You have an object which contains only a single string.
```
{
  "one": {
    "two": {
      "three": [
        "four"
      ]
    }
  }
}
```
This is the closest valid JSON to what you wrote.
```
{
  "one": {
    "two": {
      "three": {
        "four": "something"
      }
    }
  }
}
```
This is another possibility, but you didn't provide the `"something"`.

**Reply by KojoBailey** [reply]
Well, there is no `"something"`, so I guess an array is correct after all, thanks.

Although there's still the question of how to generate the objects from a loop.

**Reply by gregmarr** [reply]
I guess something like this:

```
void populateFromVector(json &j, std::vector<std::string> const &vec)
{
    auto jcur = &j;

    for (auto it = vec.begin(); it < vec.end() - 1; it++)
    {
        jcur = &(*jcur)[*it];
    }

    jcur->push_back(vec.back());
}
```
